src/thread.cr

```diff
@@ -25,14 +26,22 @@ class Thread
   def initialize(&@func : ->)
     @th = uninitialized LibC::PthreadT
 
-    ret = GC.pthread_create(pointerof(@th), Pointer(LibC::PthreadAttrT).null, ->(data : Void*) {
+    ret = LibC.pthread_attr_init(out attr)
+    raise Errno.new("pthread_attr_init", ret) unless ret == 0
+    @attr = attr
+
+    ret = GC.pthread_create(pointerof(@th), pointerof(attr), ->(data : Void*) {
       (data.as(Thread)).start
       Pointer(Void).null
     }, self.as(Void*))
 
     if ret == 0
       @@threads.push(self)
     else
+      @attr.try do |attr|
```

Comment:
We already have non-nillable `attr` (not `@attr`) above.

Reply:
Ha! Silly me for copy/paste of my code from the finalizer.